File: ObjectAvoidance/java/src/main/java/elite/math/trig/Arccosecant.java
package elite.math.trig;

import elite.math.trig.triangle.Hypotenuse;
import elite.math.trig.triangle.Opposite;

/**
 *
 * @author CFerg (Elite)
 */
public class Arccosecant {

    /**
     *
     */
    public double arccosecant;
    
    /**
     *
     * @param o
     * @param h
     */
    public Arccosecant(Opposite o, Hypotenuse h){
        this.arccosecant = (o.opposite / h.hypotenuse);
    }
    
    /**
     *
     * @param csc
     */
    public Arccosecant(Cosecant csc){
        this.arccosecant = (1 / csc.cosecant);
    }
    
    /**
     *
     * @param sin
     */
    public Arccosecant(Sine sin){
        this.arccosecant = sin.sine;
    }
}

File: ObjectAvoidance/src/elite/math/trig/Arccosecant.kt
package elite.math.trig

import elite.math.trig.triangle.Hypotenuse
import elite.math.trig.triangle.Opposite

/**
 *
 * @author CFerg (Elite)
 */
class Arccosecant {

    /**
     *
     */
    var arccosecant: Double = 0.toDouble()

    /**
     *
     * @param o
     * @param h
     */
    constructor(o: Opposite, h: Hypotenuse) {
        this.arccosecant = o.opposite / h.hypotenuse
    }

    /**
     *
     * @param csc
     */
    constructor(csc: Cosecant) {
        this.arccosecant = 1 / csc.cosecant
    }

    /**
     *
     * @param sin
     */
    constructor(sin: Sine) {
        this.arccosecant = sin.sine
    }
}